### PR TITLE
upgrade to Julia v0.7

### PR DESCRIPTION
**Pull request by RalphAS**
This works on a system with patched FileIO and ImageFiltering, so don't look too hard until those are updated and tagged.

### CODE REVIEW

**Comment by timholy** [comment]
Thanks! I worry a bit about conflicts with #123, also a very big patch. @SimonDanisch, care to chime in with your thoughts?

**Comment by RalphAS** [comment]
I naively thought that "binarybuilder" indicated a narrowly focussed update which would be easy to merge.  After looking closer, I learned otherwise.

Accordingly I have made a new branch (on my fork, not currently a PR here), standing on Simon's shoulders.  The result runs (and passes tests!) with local patches but not on the CI servers. (Some published but untagged revisions of dependencies might suffice, but I gather that it is bad form to use those in CI scripts).   Should I force-replace this PR branch with the one that incorporates Simon's commits?
@timholy @SimonDanisch

**Comment by bjarthur** [comment]
please tag a release before dropping support for julia 0.6!

**Comment by bjarthur** [comment]
PR in metadata now...  https://github.com/JuliaLang/METADATA.jl/pull/15632

**Comment by RalphAS** [comment]
I have rebased (and added a few new patches) to accommodate the merge of #121.  This is still awaiting merging of updates to FileIO and ImageFiltering.

File: .travis.yml [hunk 1]
```diff
@@ -3,18 +3,21 @@ os:
   - linux
   - osx
 julia:
-  - 0.6
+  - 0.7
+  - 1.0
   - nightly
 matrix:
   allow_failures:
+    - julia: 1.0
```
Comment (musm) [inline .travis.yml]: probably remove this

**Comment by RalphAS** [comment]
Subsumed in #125

**Pull request closed without merging**